src/istio/ambient/zarf.yaml

```diff
@@ -0,0 +1,76 @@
+# Copyright 2024 Defense Unicorns
+# SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+
+kind: ZarfPackageConfig
+metadata:
+  name: uds-core-istio-ambient
+  description: "UDS Core Istio Ambient"
+  url: https://istio.io/latest/
+variables:
+  - name: CNI_CONF_DIR
+    description: "CNI configuration directory"
+    default: ""
+  - name: CNI_BIN_DIR
+    description: "CNI binary directory"
+    default: ""
+    
+components:
+  - name: istio-ambient
+    required: true
+    charts:
+      - name: uds-istio-ambient-config
+        namespace: istio-system
+        version: 0.1.0
+        localPath: chart
+      - name: cni
+        url: https://istio-release.storage.googleapis.com/charts
+        version: 1.24.2
+        namespace: istio-system
+        valuesFiles:
+          - "../values/base-cni.yaml"
+      - name: ztunnel
+        url: https://istio-release.storage.googleapis.com/charts
+        version: 1.24.2
+        namespace: istio-system
+    actions:
+      onDeploy:
+        before:
+          - description: "Ensure CNI_CONF_DIR is set"
+            cmd: |
+              if [ \"${ZARF_VAR_CNI_CONF_DIR}\" == \"\" ]; then
+                if ./zarf tools kubectl version -o json 2>/dev/null | ./zarf tools yq '.serverVersion.gitVersion' 2>/dev/null | grep -q "k3s"; then
+                  if ./zarf tools kubectl get nodes -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}' 2>/dev/null | grep -q "k3d"; then
+                    echo "/var/lib/rancher/k3s/agent/etc/cni/net.d"
+                  fi
```

Comment:
Should this check for k3d here? Based on [the upstream profiles](https://github.com/istio/istio/tree/master/manifests/helm-profiles) I think we'd want `/var/lib/rancher/k3s/agent/etc/cni/net.d` for both k3s and k3d, so might not need the extra node check here?

Reply:
Oh yeah whoops I thought I took that out. Good catch!